posthog/queries/person_query.py

```diff
@@ -102,18 +102,23 @@ def get_query(
         )
         updated_after_clause, updated_after_params = self._get_updated_after_clause()
 
+        cohort_subquery = (
+            f"""AND id IN (
+            SELECT id FROM person
+            {cohort_query}
+            WHERE team_id = %(team_id)s
+            {person_filters}
+        )
+        {cohort_filters}"""
+            if cohort_query
+            else ""
+        )
         return (
             f"""
             SELECT {fields}
             FROM person
             WHERE team_id = %(team_id)s
-            AND id IN (
-                SELECT id FROM person
-                {cohort_query}
-                WHERE team_id = %(team_id)s
-                {person_filters}
-            )
-            {cohort_filters}
+            {cohort_subquery}
```

Comment:
A bug here:

`{cohort_filters}` are disjoint from cohort_query. (yeah I know, annoying :/ )

`cohort_query` is a optimisation that supports only a single cohort. 

`cohort_filters` support multiple cohorts, but are less efficient.

So, this should be something like:

```
{cohort_subquery}
{cohort_filters}
```

I'm sad I did not add a test case for something like this, but a valid test case here would be:

https://github.com/PostHog/posthog/pull/13161/files#diff-7d9c4d6fc0009dff4f4c3a7b27b40f6e7d1d3eb0cc634a8f0bd1621028b19aa0R1991-R1997

This test, with atleast one person property filter as well so it goes into the `if property_filters` conditional